src/viam/examples/modules/complex/summation/impl.hpp

```diff
@@ -17,6 +17,7 @@ class MySummation : public Summation {
     };
     void reconfigure(Dependencies deps, ResourceConfig cfg) override;
     static std::vector<std::string> validate(ResourceConfig cfg);
+    void stop(const AttributeMap& extra) override{};
```

Comment:
There were quite a few places I had to add `stop` due to it being a purely virtual method (I don't really understand how #182 was merged without these changes). I would say I'm against having every resource implement its own `stop` method, and would rather use a noop default method for the time being. IMO there should be an `ActuatingComponent` class or something that inherits from `Component` with virtual `is_moving` and `stop` methods, but that might be work for another ticket? Let me know if I'm missing some reason for the purely virtual `stop`.

Reply:
There's a ticket to address this [here](
https://viam.atlassian.net/browse/RSDK-5881), I think the current proposal is to have a `StoppableResource` class that can be inherited from. 

I understand the inconvenience of having to implement a no-op `stop` for sure but I'm not convinced that having `stop` default is better. It's (trivially) less work and a bit less pointy, but it also makes it possible for people to potentially do Quite Bad things by failing to implement `stop` when it _is_ necessary. My preference would be to back out the `stop` changes here and just focus on the `ResourceType` removal, and then make the more principled change in a separate PR.

Reply:
My bad; totally missed that ticket! Totally understand the concern; I'll remove all `stop` changes from this PR for now, thanks for the context.